Python/bytecodes.c

```diff
@@ -177,12 +176,9 @@ dummy_func(
             }
         }
 
-        inst(LOAD_CLOSURE, (-- value)) {
-            /* We keep LOAD_CLOSURE so that the bytecode stays more readable. */
-            value = GETLOCAL(oparg);
-            ERROR_IF(value == NULL, unbound_local_error);
```

Comment:
I note that we are implicitly eliminating this NULL check by converting `LOAD_CLOSURE` to `LOAD_FAST`. I _think_ that this is actually just fine, because ensuring this local exists (and is a cell variable) is the responsibility of the compiler, not the code author. The compiler will always insert a `MAKE_CELL` at the top of the function for any cell variable, so I can't see any way this null check would ever have failed. If it ever did, that would indicate a compiler bug, so it's not clear that `UnboundLocalError` would ever have made sense here anyway.

Reply:
@markshannon What do you think? We could replace it with LOAD_CLOSURE_CHECK instead to preserve this. There might be a way for a debugger to delete the variable.

Reply:
> There might be a way for a debugger to delete the variable.

I don't believe there is, unless the debugger is doing something unsupported and poking directly at the fastlocals array on a `_PyInterpreterFrame`. The supported way for debuggers to change the value of variables is via `frame.f_locals` dict, which is synced back to fast locals by `PyFrame_LocalsToFast`, and will only update cell contents, never overwrite the cell itself.